spec.html

```diff
@@ -1096,34 +1097,31 @@ <h1>DecorateClass ( elements, decorators )</h1>
     </emu-clause>
 
     <emu-clause id=sec-decorate-element aoid=DecorateElement>
-      <h1>DecorateElement ( element, keys )</h1>
-      <p>With parameters _element_, a Class Element and _keys_, a List of Property Keys and Private Names.</p>
+      <h1>DecorateElement ( _element_, _placements_ )</h1>
+      <p>With parameters _element_, a Class Element, and _placements_:</p>
       <emu-alg>
         1. Let _extras_ be a new empty List.
         1. Let _finishers_ be a new empty List.
-        1. For each _decorator_ in element_.[[Decorators]], in reverse list order do
-          1. Assert: _element_.[[Key]] is an element of _keys_.
-          1. Remove _element_.[[Key]] from _keys_.
+        1. For each _decorator_ in _element_.[[Decorators]], in reverse list order do
+          1. Perform RemoveElementPlacement(_element_, _placements_).
```

Comment:
This process of removing and re-adding the elements will change the enumeration order, e.g. extras may precede the element. Is this intentional?

Reply:
This doesn't change order at all. placements are just bags of property names.